src/lib/is-hostname.js

```diff
@@ -0,0 +1,11 @@
+// checks whether the hostname has been entered into the format that Nori expects, because Nori then interpolates it into the string it uses when calling Tako.
+
+const isHostname = hostname => {
+	// check if totally mad
+	if (/(https:\/\/)*(.*)\/tako(.*)/.test(hostname)) {
+		return /(https:\/\/)*(.*)\/tako(.*)/g.exec(hostname)[2]
```

Comment:
returning this extracted bit from `validate` will surface it as a validation error. there's a separate option called `result` to do this kind of coercion. you'd probably want to extract the regex as a toplevel variable

Reply:
Weirdly when you extract the regex as a top level variable it gets consumed somehow and then is unusable on line 6.

Reply:
woah TIL https://stackoverflow.com/a/11477448/805227